### PR TITLE
rootless: use slirp4netns to setup the network namespace

### PR DESCRIPTION
**Pull request by giuseppe**
If slirp4netns is available, use it to configure the network for the
rootless isolation mode.

Closes: https://github.com/containers/buildah/issues/1139

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: run.go [hunk 1]
```diff
@@ -1639,9 +1643,68 @@ func runCollectOutput(fds, closeBeforeReadingFds []int) string {
...
+	if err != nil {
+		return nil, errors.Wrapf(err, "cannot start slirp4netns")
+	}
+
+	b := make([]byte, 1)
+	if _, err := rootlessSlirpSyncR.Read(b); err != nil {
+		return nil, errors.Wrapf(err, "cannot read from slirp4netns sync pipe")
+	}
+
+	go cmd.Wait()
```
Comment (AkihiroSuda) [inline run.go]: ?

Comment (giuseppe) [reply run.go]: that is for cleaning up, I've moved it to the teardown function.  Could you please check if the new version is better?

**Comment by TomSweeneyRedHat** [comment]
LGTM assuming happy tests.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Comment by giuseppe** [comment]
@rh-atomic-bot retry

**Comment by giuseppe** [comment]
@rh-atomic-bot retry

**Comment by giuseppe** [comment]
@rh-atomic-bot r=rhatdan

**Pull request closed without merging**